### PR TITLE
Feature/ast 11451 add empty projects

### PR DESCRIPTION
**Pull request by serhii-stanislavskyi**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../CODE-OF-CONDUCT.md). Please see the [contributing guidelines](../CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: internal/process.go [hunk 1]
```diff
@@ -300,6 +304,18 @@ func fetchTeamsData(client rest.Client, exporter export2.Exporter) error {
 	return nil
 }
 
+func fetchProjectsData(client rest.Client, exporter export2.Exporter) error {
```
Comment (constantinoantunes1) [inline internal/process.go:307]: Please add unit tests

File: internal/integration/rest/models.go [hunk 2]
```diff
@@ -63,6 +63,26 @@ type (
 		ParendID int    `json:"parentId"`
 	}
 
+	Project struct {
+		ID                 int                   `json:"id"`
+		TeamID             int                   `json:"teamId"`
+		Name               string                `json:"name"`
+		IsPublic           bool                  `json:"isPublic"`
+		SourceSettingsLink SourceSettingsLinkObj `json:"sourceSettingsLink"`
+		Link               LinkObj               `json:"link"`
```
Comment (constantinoantunes1) [inline internal/integration/rest/models.go:72]: The link attribute should be for API navigation only, and not needed for the export.

File: internal/integration/rest/models.go [hunk 3]
```diff
@@ -63,6 +63,26 @@ type (
 		ParendID int    `json:"parentId"`
 	}
 
+	Project struct {
+		ID                 int                   `json:"id"`
+		TeamID             int                   `json:"teamId"`
+		Name               string                `json:"name"`
+		IsPublic           bool                  `json:"isPublic"`
+		SourceSettingsLink SourceSettingsLinkObj `json:"sourceSettingsLink"`
```
Comment (constantinoantunes1) [inline internal/integration/rest/models.go:71]: From [API docs](https://checkmarx.atlassian.net/wiki/spaces/SAST/pages/3206357767/CxSAST+REST+API+v1) it seems this is an API link to another resource, and doesn't belong to the export.
Instead, we could use the link to find the sourceSettings object, and then include it in the export.

File: Makefile [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 PRODUCT_NAME = cxsast_exporter
 PRODUCT_VERSION = $(shell cat VERSION)
 PRODUCT_BUILD = $(shell date +%Y%m%d%H%M%S)
 PUBLIC_KEY = "internal/app/encryption/public.key"
 LD_FLAGS = -ldflags="-s -w -X github.com/checkmarxDev/ast-sast-export/cmd.productName=$(PRODUCT_NAME) -X github.com/checkmarxDev/ast-sast-export/cmd.productVersion=$(PRODUCT_VERSION) -X github.com/checkmarxDev/ast-sast-export/cmd.productBuild=$(PRODUCT_BUILD)"
 
-SAST_EXPORT_USER = '###########'
-SAST_EXPORT_PASS = '###########'
+SAST_EXPORT_USER = 'ssd'
+SAST_EXPORT_PASS = 'Cx12345678!'
```
Comment (ecasanova-checkmarx) [inline Makefile:12]: Remove this please.

**Review by ecasanova-checkmarx — CHANGES_REQUESTED** [review]

**Review by constantinoantunes1 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 4 above)
Comment (serhii-stanislavskyi) [reply Makefile]: Removed

File: internal/process.go (hunk 1 above)
Comment (serhii-stanislavskyi) [reply internal/process.go:307]: I'll add

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (serhii-stanislavskyi) [reply Makefile]: Removed

File: Makefile [hunk 5]
```diff
@@ -1,15 +1,15 @@
 KEYS_PATH = ./keys
 EXTERNAL_PATH = ./external
 BUILD_PATH = ./build
-ENV ?= prod
+ENV ?= dev
```
[the remark this answers is not in the document]
Comment (serhii-stanislavskyi) [reply Makefile]: Fixed

File: internal/integration/rest/models.go (hunk 3 above)
Comment (serhii-stanislavskyi) [reply internal/integration/rest/models.go:71]: Do you mean to take link from here?
`"link": {
      "rel": "string",
      "uri": "string"
    }`
From uri field

**Comment by fjsnogueira** [comment]
Logo
**Checkmarx AST – Scan Summary & Details** – [92667052\-f1cc\-4afc\-87bb\-b9fdec406076](https://eu.ast.checkmarx.net/projects/e03e84ae-72f8-46a3-9063-082d729bfb5f/scans?id=92667052-f1cc-4afc-87bb-b9fdec406076&branch=feature%2FAST-11451-add-empty-projects)
### No New Or Fixed Issues Found

**Review by ecasanova-checkmarx — APPROVED** [review]

File: internal/integration/rest/models.go [hunk 6]
```diff
@@ -63,6 +63,13 @@ type (
 		ParendID int    `json:"parentId"`
 	}
 
+	Project struct {
+		ID       int    `json:"id"`
+		TeamID   int    `json:"teamId"`
+		Name     string `json:"name"`
+		IsPublic bool   `json:"isPublic"`
```
Comment (constantinoantunes1) [inline internal/integration/rest/models.go:70]: Please confirm with Wombat team these are the fields they need for project import in AST.

**Review by constantinoantunes1 — APPROVED** [review]

**Pull request merged**